dash/development/_r_components_generation.py

```diff
@@ -425,6 +446,7 @@ def write_js_metadata(pkg_data, project_shortname):
     file_path = os.path.join("R", file_name)
     with open(file_path, "w") as f:
         f.write(function_string)
+        f.write(component_helpers)
```

Comment:
`if has_wildcards`?

Reply:
Good catch. Will fix.

Reply:
fixed in c6a013a15eb90a5cc1cf939d4f711133800bd255

Reply:
Somehow this change (only adding `dash_assert_valid_wildcards` to `internal.R` if needed by a component) was reverted - was that intended?

Reply:
Yes. It is preferable not to generate this function for all generated component libraries, but I am uncertain about how to check for wildcards without passing props to methods that ordinarily would not require them. 

Any advice you have on how to do this cleanly without refactoring the current code would be helpful.

Reply:
I just looped through the metadata for wildcards again -> https://github.com/plotly/dash/pull/783/commits/15cbea84e4be40125b7ef3302de32ee6ec0fa92f